src/interfaces.jl

```diff
@@ -110,3 +122,27 @@ function reverse_time(sol::Solver)
     options.stopevent = (t,y)->stopevent(2*t0-t,y)
     return solve(ode_reversed,stepper,options)
 end
+
+
+
+# The the elements of tspan should basically be scalars and support
+# most of the scalar operations.  In particular the element type
+# should be closed under the division.
+function make_consistent{S<:AbstractStepper}(y0, tspan::AbstractVector, stepper::Type{S})
+    t_test = 1/(tspan[end]-tspan[1])
+    Tt = typeof(t_test)
+    t_new = convert(Vector{Tt},tspan)
+
+    y_test = y0./t_test
+    Ty = typeof(y_test)
+    if typejoin(Ty,AbstractArray) == AbstractArray
+        Ey = promote_type(map(typeof,y_test)...)
```

Comment:
I thought that maybe someone does indeed need to do a calculation with a `Any[]` or `FloatingPoint[]`, but in most cases it would be a mistake.  Thus the warning but not an error nor a conversion.